v3/src/data-interactive/handlers/color-map-handler.ts

```diff
@@ -0,0 +1,87 @@
+import { createProvisionalCategorySet } from "../../models/data/category-set"
+import { getSharedCaseMetadataFromDataset } from "../../models/shared/shared-data-utils"
+import { DICategoryColorMap } from "../data-interactive-data-set-types"
+import { registerDIHandler } from "../data-interactive-handler"
+import { DIHandler, DIResources, DIValues } from "../data-interactive-types"
+import {
+  attributeNotFoundResult, collectionNotFoundResult, dataContextNotFoundResult, errorResult,
+} from "./di-results"
+
+export const diCategoryColorMapHandler: DIHandler = {
+  create(resources: DIResources, _values?: DIValues & { metadata?: any }) {
+    const { dataContext, collection, attribute } = resources
+    if (!dataContext) return dataContextNotFoundResult
```

Comment:
You should check for `collection` and `attribute` here as well, like you do in `get`. There's no need to do anything else if we know the handler will fail. It's also easier to engineers to follow to group these things together.